### DISCUSSION TITLE
Fungsi yang harus digunakan

### DISCUSSION THREAD

**Discussion by Anggitnoviee** [teamalgoritma/community#206] [Q&A] [answered]
Hallo saya mempunyai kolom berisi [ P22.1;P03.4;Z38.0 ] dan ingin menghapus [;P03.4;Z38.0 ] itu bagaimana? biar output yang dihasilkan hanya P22.1 dengan menggunana R

**Comment by triarm** [comment, marked as answer]
Hi @Anggitnoviee ,
terkait hal tersebut Anda dapat melakukan subsetting terhadap kolom yang ingin Anda ambil. Misalkan data tersebut Anda simpan dalam object bernama `df`. Jika Anda hanya ingin mengambil kolom "P22.1" maka Anda dapat melakukan subsetting sebagai berikut

```
df[,"P22.1"]
```

Semoga menjawab

**Reply by Anggitnoviee** [reply]
kalo semua kolom sampe bawah, harus seperti apa kak @triarm

**Reply by triarm** [reply]
Apakah maksudnya semua baris sampai ke bawah yang diambil bu? Jika ya, dengan kode di atas seharusnya semua baris terambil.

**Reply by Anggitnoviee** [reply]
Baris selanjutnya itu beda2 kak
Kolom A
P22.1;P03.4;Z38.0
P33.7
Q21.0;K03.9
.
.
dst
Nah dari data tersebut saya hanya ingin mengambil depannyanya jadi yg dibelakang ; dihilangkan bgitu kak, kira2 gimana ya kak🙏🏻 Mohon maaf sebelumnya @triarm

**Reply by Anggitnoviee** [reply]
bagaiaman kak🙏🏻 @inytss @triarm

**Reply by triarm** [reply]
Hi @Anggitnoviee ,
Sebelumnya saya ingin memastikan apakah kolom P22.1;P03,4;Z38.0 tersebut memang berada pada satu kolom atau seharusnya berada pada kolom berbeda? Anda dapat mencoba membaca data kembali namun tambahkan parameter `sep = ";"`

Semoga menjawab.

**Reply by Anggitnoviee** [reply]
ketika menggunakan parameter sep ";" data erorr tidak mau terbaca, data tersebut memang berada dalam satu kolom dan hanya ingin diambil depannya saja🙏🏻 @triarm

**Reply by triarm** [reply]
Kalau seperti itu, Anda dapat menjalankan kode berikut untuk mengekstrak teks sebelum tanda ;
```
gsub("\\;.*", "", df$kolomA)
```

**Reply by Anggitnoviee** [reply]
Terimakasih kak sudah bisa
